service/controller/resource/clusterconfigmap/desired.go

```diff
@@ -125,84 +120,99 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) ([]*cor
 					Kind:    infrastructureRef.Kind,
 					Version: infrastructureRef.GroupVersionKind().Version,
 				})
-				err = r.k8sClient.CtrlClient().Get(context.Background(), client.ObjectKey{
-					Namespace: infrastructureRef.Namespace,
+				err = r.k8sClient.CtrlClient().Get(ctx, client.ObjectKey{
+					Namespace: cr.Namespace,
 					Name:      infrastructureRef.Name,
 				}, gcpCluster)
 				if err != nil {
 					return nil, microerror.Mask(err)
 				}
 
-				gcpProject, _, err := unstructured.NestedString(gcpCluster.Object, []string{"spec", "project"}...)
+				gcpProject, _, err = unstructured.NestedString(gcpCluster.Object, []string{"spec", "project"}...)
 				if err != nil {
 					return nil, microerror.Mask(err)
 				}
-				clusterValues["gcpProject"] = gcpProject
-
-				region, _, err := unstructured.NestedString(gcpCluster.Object, []string{"spec", "region"}...)
-				if err != nil {
-					return nil, microerror.Mask(err)
-				}
-				clusterValues["region"] = region
-
-			case "OpenStackCluster":
-				clusterValues["provider"] = "openstack"
-
-			case "VSphereCluster":
-				clusterValues["provider"] = "vsphere"
-
 			default:
-				r.logger.Debugf(ctx, "unable to extract infrastructure provider-specific clusterValues for cluster. Unsupported infrastructure kind %q", infrastructureRef.Kind)
+				r.logger.Debugf(ctx, "unable to extract infrastructure provider-specific clusterValues for cluster. Unsupported infrastructure kind %q", r.provider)
 			}
 		}
 	}
 
-	configMapSpecs := []configMapSpec{
-		{
-			Name:      key.AppOperatorValuesResourceName(&cr),
-			Namespace: cr.GetNamespace(),
-			Values:    appOperatorValues,
+	appOperatorValues := map[string]interface{}{
+		"app": map[string]interface{}{
+			"watchNamespace":    cr.GetNamespace(),
+			"workloadClusterID": key.ClusterID(&cr),
 		},
-		{
-			Name:      key.ClusterValuesResourceName(&cr),
-			Namespace: cr.GetNamespace(),
-			Values:    clusterValues,
+		"provider": map[string]interface{}{
+			"kind": r.provider,
+		},
+		"registry": map[string]interface{}{
+			"domain": r.registryDomain,
 		},
 	}
 
-	for _, spec := range configMapSpecs {
-		configMap, err := newConfigMap(cr, spec)
-		if err != nil {
-			return nil, microerror.Mask(err)
-		}
+	appValuesYaml, err := yaml.Marshal(appOperatorValues)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
 
-		configMaps = append(configMaps, configMap)
+	appValuesConfigMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      key.AppOperatorValuesResourceName(&cr),
+			Namespace: cr.GetNamespace(),
+			Annotations: map[string]string{
+				annotation.Notes: fmt.Sprintf("DO NOT EDIT. Values managed by %s.", project.Name()),
+			},
+			Labels: map[string]string{
+				label.Cluster:   key.ClusterID(&cr),
+				label.ManagedBy: project.Name(),
+			},
+		},
+		Data: map[string]string{
+			"values": string(appValuesYaml),
+		},
 	}
 
-	return configMaps, nil
-}
+	clusterValues := ClusterValuesConfig{
```

Comment:
Instead of using `map[string]interface{}` I made a typed struct for the cluster values. It should work the same way, but it's easier to work with and test.